packages/main/src/SegmentedButton.ts

```diff
@@ -172,19 +197,18 @@ class SegmentedButton extends UI5Element {
 	}
 
 	normalizeSelection() {
-		const selectedItems = this.items.filter(item => item.pressed);
-		const selectedIndex = this._selectedItem ? selectedItems.indexOf(this._selectedItem) : -1;
-
-		if (this._selectedItem && selectedItems.length > 1) {
-			selectedItems.splice(selectedIndex, 1);
+		switch (this.mode) {
+		case SegmentedButtonMode.Single: {
+			const pressedItems = this.pressedItems;
+			const pressedItemIndex = this._pressedItem ? pressedItems.indexOf(this._pressedItem) : -1;
+			if (this._pressedItem && pressedItems.length > 1) {
+				pressedItems.splice(pressedItemIndex, 1);
+			}
+			const pressedItem = pressedItems.pop() || this.items[0];
```

Comment:
@unazko Thank you for applying the previous review suggestions.
I am only wondering about this line :
`const pressedItem = pressedItems.pop() || this.items[0];` and the addition of `this.items[0]`. It was not there before and the entire section is about "Single" selection. Is it required and when. Thanks

Reply:
The `pressedItems.length` could be `0`. In this case we would have nothing to pop out and the only logical choice for selection we have would be the first segmented button item. That is only possible when setting the item's pressed state programmatically and not with user interaction. The `_applySingleSelection` method would always need an item to apply the single selection to as we do not have an scenario without a pressed item in this mode.